### PR TITLE
rename samples

### PR DESCRIPTION
**Pull request by JeanTracker**
gui: gui sample application
    extension: extension sample for standalone application
    daemon: daemon sample for multi application

Signed-off-by: JeanTracker <hyojoong.kim.jean@gmail.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 3.9)
+PROJECT(nugu-examples)
 
-ADD_SUBDIRECTORY(single-app)
-ADD_SUBDIRECTORY(multi-app)
-ADD_SUBDIRECTORY(gui-app)
\ No newline at end of file
+ADD_SUBDIRECTORY(extension)
+ADD_SUBDIRECTORY(daemon)
+ADD_SUBDIRECTORY(gui)
```
Comment (kimhyungrok) [inline CMakeLists.txt:6]: It recommend to add line break.

File: daemon/main.cc [hunk 2]
```diff
@@ -2,6 +2,6 @@
 
 int main(int argc, char* args[])
 {
-    std::cout << "single application example" << std::endl;
+    std::cout << "daemon sample for multi application" << std::endl;
     return 1;
 }
```
Comment (kimhyungrok) [inline daemon/main.cc:7]: It recommend to add line break.

File: extension/main.cc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+
+int main(int argc, char* args[])
+{
+    std::cout << "extension sample for standalone application" << std::endl;
+    return 1;
+}
```
Comment (kimhyungrok) [inline extension/main.cc:7]: It recommend to add line break.

File: CMakeLists.txt (hunk 1 above)
Comment (JeanTracker) [reply CMakeLists.txt]: I fixed it. thanks :)

File: daemon/main.cc (hunk 2 above)
Comment (JeanTracker) [reply daemon/main.cc]: I fixed it. thanks :)

File: extension/main.cc (hunk 3 above)
Comment (JeanTracker) [reply extension/main.cc]: I fixed it. thanks :)

**Review by kimhyungrok — APPROVED** [review]

**Pull request merged**